include/clangmetatool/collectors/include_graph.h

```diff
@@ -46,6 +46,30 @@ class IncludeGraph {
    * Get the pointer to the data structure, populated or not.
    */
   IncludeGraphData *getData();
+
+  /**
+   * Decrement the reference count tracked for this edge, notifying the caller
+   * if the operation was successful.
+   *
+   * Return a boolean if indicating we were able to decrement the count:
+   *  - false if the edge doesn't exist or has no balance to decrement
+   *  - true otherwise. This will always reduce the count by 1
+   */
+  bool decrementRefCount(const clangmetatool::types::FileGraphEdge &edge);
```

Comment:
I don't think we should have functions here that operate on the data held by the collector. Instead, users can copy it out and modify it if needed. Maybe these functions should then be on the `IncludeGraphData` object itself?

Reply:
Agreed, will move this to be defined on the struct instead, or would a free function accepting a pointer be okay too?